### PR TITLE
Harsh/geohashes practice

### LINKED ISSUE #107: Feature - [Doctor] Add fees and location field in profile screen
**Is your feature request related to a problem? Please describe.**
- Add fees field in profile ui as well store data in firebase user_data table

### Check List

- [x] I have linked my PR with this task
- [x] I have added comment after completing this task
- [x] I have followed all [guidelines](https://github.com/jaykishan-sewak/doctor-appoint-app/tree/main) for code

### LINKED ISSUE #173: Bug - [Doctor] Update profile screen app crash
**Describe the bug**
- Doctor first time move to Update profile screen and fill data that time app is crash.

**To Reproduce**
1. New doctor register
2. login it
3. Doctor move to update profile screen
4. First click on email verify
5. add Address
6. Not verify phone
7. Add other fields data and time slot add more then one
8. Now verify phone
9. now select image from gallery
10. Now click on update api call and app is crash.

Device information
- Samsung

### Check List

- [x] I have linked my PR with this task
- [x] I have added comment after completing this task
- [x] I have followed all [guidelines](https://github.com/jaykishan-sewak/doctor-appoint-app/tree/main) for code

### PR DESCRIPTION
**Pull request by HarshPanchalInfusion**
Task Completed

### CODE REVIEW

File: app/src/main/java/com/android/doctorapp/repository/AdminRepository.kt [hunk 1]
```diff
@@ -46,13 +55,86 @@ class AdminRepository @Inject constructor(
...
+        longitude: Double
+    ): ApiResponse<List<UserDataResponseModel>> {
+        return try {
+            val center = GeoLocation(latitude, longitude)
+            val radiusInM = 50.0 * 1000.0
+            val bounds = GeoFireUtils.getGeoHashQueryBounds(center, radiusInM)
+            val doctorTasks: MutableList<Task<QuerySnapshot>> = ArrayList()
+            val userList = arrayListOf<UserDataResponseModel>()
+            var feedback = FeedbackResponseModel()
+//            for (b in bounds) {
```
Comment (vikas-infusion) [inline app/src/main/java/com/android/doctorapp/repository/AdminRepository.kt]: Remove commented code

File: app/src/main/java/com/android/doctorapp/repository/AuthRepository.kt [hunk 2]
```diff
@@ -216,6 +218,51 @@ class AuthRepository @Inject constructor(
         }
     }
 
+    suspend fun testGeoHash(
+        doctorRequestModel: UserDataRequestModel,
+        fireStore: FirebaseFirestore
+    ): ApiResponse<UserDataRequestModel> {
+        return try {
+            /*val latitude = 23.0225
```
Comment (vikas-infusion) [inline app/src/main/java/com/android/doctorapp/repository/AuthRepository.kt]: Remove commented code

File: app/src/main/java/com/android/doctorapp/repository/models/UserDataResponseModel.kt [hunk 3]
```diff
@@ -11,6 +14,36 @@ data class UserDataResponseModel(
...
+    var weekOffList: ArrayList<String>? = null,
+    var rating: Float? = null,
+    var feedbackDetails: FeedbackResponseModel? = null,
+    var addressLatLng: Map<String, Any>? = null,
+    var geohash: String? = null
+
+
+
+/*
+var id: String = "",
```
Comment (vikas-infusion) [inline app/src/main/java/com/android/doctorapp/repository/models/UserDataResponseModel.kt]: Remove commented

File: app/src/main/java/com/android/doctorapp/ui/authentication/login/LoginViewModel.kt [hunk 4]
```diff
@@ -49,10 +49,10 @@ class LoginViewModel @Inject constructor(
 
     var googleSignInClient: GoogleSignInClient
 
-    val email: MutableLiveData<String> = MutableLiveData()
+    val email: MutableLiveData<String> = MutableLiveData("myuser1@mailinator.com")
```
Comment (vikas-infusion) [inline app/src/main/java/com/android/doctorapp/ui/authentication/login/LoginViewModel.kt]: Remove static credentials

File: app/src/main/java/com/android/doctorapp/ui/doctor/AddDoctorViewModel.kt [hunk 5]
```diff
@@ -414,6 +424,137 @@ class AddDoctorViewModel @Inject constructor(
...
+            /*when (val response = authRepository.testGeoHash(fireStore)) {
+                else -> {
+
+                }
+            }*/
+
+            val latitude = 23.0225
+            val longitude = 72.5714
+
+            val geohash = GeoFireUtils.getGeoHashForLocation(GeoLocation(latitude, longitude))
```
Comment (vikas-infusion) [inline app/src/main/java/com/android/doctorapp/ui/doctor/AddDoctorViewModel.kt]: Please remove unused variable and remove all commented code in this task

File: app/src/main/java/com/android/doctorapp/ui/doctordashboard/doctorfragment/DoctorAddressFragment.kt [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+                viewModel.addressLatLngList.value = updates
+//                Log.d("TAG", "onLocationResult 156 : $updates")
+
+
+                findNavController().previousBackStackEntry?.savedStateHandle?.set("address", address)
+                findNavController().previousBackStackEntry?.savedStateHandle?.set("addressLatLng", updates)
+                fusedLocationClient.removeLocationUpdates(this)
+                findNavController().popBackStack()
+
+                // Find cities within 50km of London
```
Comment (vikas-infusion) [inline app/src/main/java/com/android/doctorapp/ui/doctordashboard/doctorfragment/DoctorAddressFragment.kt]: Please remove all commented code and check all things that need to be reduce code wise.

File: app/src/main/res/layout/fragment_doctor_address.xml [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <androidx.appcompat.widget.AppCompatTextView
+            android:id="@+id/text_use_current_location"
+            android:layout_width="match_parent"
+            android:layout_height="@dimen/margin_50"
+            android:gravity="center"
+            android:text="Use My Current Location"
```
Comment (vikas-infusion) [inline app/src/main/res/layout/fragment_doctor_address.xml]: add this text in string.xml file

File: app/src/main/res/layout/fragment_update_doctor_profile.xml [hunk 8]
```diff
@@ -129,6 +130,17 @@
 
                 </com.google.android.material.textfield.TextInputLayout>
 
+                <androidx.appcompat.widget.AppCompatTextView
+                    android:id="@+id/text_address"
+                    style="@style/custom_edit_text_bg"
+                    android:hint="Enter address"
```
Comment (vikas-infusion) [inline app/src/main/res/layout/fragment_update_doctor_profile.xml]: add this text in string.xml file

File: app/src/main/res/layout/fragment_update_doctor_profile.xml [hunk 9]
```diff
@@ -260,6 +272,37 @@
...
+                        android:afterTextChanged="@{(text) -> viewModel.isValidFees(text)}"
+                        android:hint="@string/hint_doctor_fees"
+                        android:imeOptions="actionNext"
+                        android:inputType="number"
+                        android:singleLine="true"
+                        android:text="@={viewModel.fees}"
+                        android:textColor="@color/black"
+                        android:textSize="14sp" />
+
+                    <!--
```
Comment (vikas-infusion) [inline app/src/main/res/layout/fragment_update_doctor_profile.xml]: Remove commented code

File: app/src/main/java/com/android/doctorapp/repository/AdminRepository.kt (hunk 1 above)
Comment (HarshPanchalInfusion) [reply app/src/main/java/com/android/doctorapp/repository/AdminRepository.kt]: Done

File: app/src/main/java/com/android/doctorapp/repository/AuthRepository.kt (hunk 2 above)
Comment (HarshPanchalInfusion) [reply app/src/main/java/com/android/doctorapp/repository/AuthRepository.kt]: Done

File: app/src/main/java/com/android/doctorapp/repository/models/UserDataResponseModel.kt (hunk 3 above)
Comment (HarshPanchalInfusion) [reply app/src/main/java/com/android/doctorapp/repository/models/UserDataResponseModel.kt]: Done

File: app/src/main/java/com/android/doctorapp/ui/authentication/login/LoginViewModel.kt (hunk 4 above)
Comment (HarshPanchalInfusion) [reply app/src/main/java/com/android/doctorapp/ui/authentication/login/LoginViewModel.kt]: Done

File: app/src/main/java/com/android/doctorapp/ui/doctor/AddDoctorViewModel.kt (hunk 5 above)
Comment (HarshPanchalInfusion) [reply app/src/main/java/com/android/doctorapp/ui/doctor/AddDoctorViewModel.kt]: Done

File: app/src/main/java/com/android/doctorapp/ui/doctordashboard/doctorfragment/DoctorAddressFragment.kt (hunk 6 above)
Comment (HarshPanchalInfusion) [reply app/src/main/java/com/android/doctorapp/ui/doctordashboard/doctorfragment/DoctorAddressFragment.kt]: Done

File: app/src/main/res/layout/fragment_doctor_address.xml (hunk 7 above)
Comment (HarshPanchalInfusion) [reply app/src/main/res/layout/fragment_doctor_address.xml]: Done

File: app/src/main/res/layout/fragment_update_doctor_profile.xml (hunk 8 above)
Comment (HarshPanchalInfusion) [reply app/src/main/res/layout/fragment_update_doctor_profile.xml]: Done

File: app/src/main/res/layout/fragment_update_doctor_profile.xml (hunk 9 above)
Comment (HarshPanchalInfusion) [reply app/src/main/res/layout/fragment_update_doctor_profile.xml]: Done

**Review by vikas-infusion — APPROVED** [review]

**Review by vikas-infusion — APPROVED** [review]

**Pull request merged**